src/java/org/openzal/zal/MailboxData.java

```diff
@@ -1,18 +1,55 @@
 package org.openzal.zal;
 
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+
 public class MailboxData
 {
-  private final int            mId;
-  private final int            mSchemaGroupId;
-  private final String         mAccountId;
-  private final short          mIndexVolumeId;
+  private final Integer mId;
+  private final int mSchemaGroupId;
+  private final String mAccountId;
+  private final Short mIndexVolumeId;
+
+  public MailboxData(@Nonnull Integer mailboxid)
+  {
+    this(
+      mailboxid,
+      (short) (((mailboxid - 1) % 100) + 1),
```

Comment:
It is better to create a method (static if it can be use outside the class) to calculate it?